src/main/kotlin/gg/skytils/skytilsmod/utils/ElementaUtil.kt

```diff
@@ -42,6 +43,19 @@ fun UITextInput.limitToNumericalCharacters() = apply {
     }
 }
 
+/**
+ * Sets this [UITextInput] to [invalidColor] when a non-numeric value is entered and focus is lost.
+ * Resets to the default color when the input is made valid.
+ */
+fun UITextInput.colorIfNumeric(validColor: Color, invalidColor: Color) = apply {
+    onKeyType { _, _ ->
+        if (getText().isInteger()) setColor(validColor)
+    }
+    onFocusLost {
+        if (!getText().isInteger()) setColor(invalidColor)
+    }
+}
+
```

Comment:
Why not make it also do invalid as you're typing?

Reply:
I thought it might be annoying if you're trying to type "-x" and it briefly shows up as red—maybe that's alright?

Reply:
I mean it would make sense when it's only the `-` sign.

Reply:
Fair point, updated to show invalid inputs immediately (also attempted to squash but may have failed, unsure).